paas_app_charmer/charm.py

```diff
@@ -173,7 +175,7 @@ def get_framework_config(self) -> BaseModel:
         """
         # Will raise an AttributeError if it the attribute framework_config_class does not exist.
         framework_config_class = self.framework_config_class
-        config = dict(self.config.items())
+        config = {k: config_get_3(self, k) for k in self.config.keys()}
```

Comment:
The proposal is to retrieve the config and secret values in `charm.py` and pass it here already retrieved

Also let's pass in the secret object here rather than the values already fetched

Reply:
Updated, thanks!